packaging/rpm/init.d/grafana-server

```diff
@@ -1,155 +1,130 @@
-#! /usr/bin/env bash
-
+#!/bin/bash
+#
+# grafana-server        Startup script for Grafana
+#
 # chkconfig: 2345 80 05
-# description: Grafana web server & backend
+# description: Grafana web server and backend
 # processname: grafana
 # config: /etc/grafana/grafana.ini
 # pidfile: /var/run/grafana.pid
-
+#
 ### BEGIN INIT INFO
 # Provides:          grafana
 # Required-Start:    $all
 # Required-Stop:     $remote_fs $syslog
 # Default-Start:     2 3 4 5
 # Default-Stop:      0 1 6
-# Short-Description: Start grafana at boot time
+# Short-Description: start and stop grafana-server
 ### END INIT INFO
 
-#  tested on
-#  1. New lsb that define start-stop-daemon
-#  3. Centos with initscripts package installed
+# Source function library
+. /etc/rc.d/init.d/functions
 
+# Set the default values for grafana-server variables
 PATH=/bin:/usr/bin:/sbin:/usr/sbin
-NAME=grafana-server
-DESC="Grafana Server"
-
+PROG=grafana-server
 GRAFANA_USER=grafana
 GRAFANA_GROUP=grafana
 GRAFANA_HOME=/usr/share/grafana
 CONF_DIR=/etc/grafana
-WORK_DIR=$GRAFANA_HOME
 DATA_DIR=/var/lib/grafana
 PLUGINS_DIR=/var/lib/grafana/plugins
 LOG_DIR=/var/log/grafana
 CONF_FILE=$CONF_DIR/grafana.ini
 MAX_OPEN_FILES=10000
-PID_FILE=/var/run/$NAME.pid
-DAEMON=/usr/sbin/$NAME
-
-if [ `id -u` -ne 0 ]; then
-  echo "You need root privileges to run this script"
-  exit 4
-fi
-
-if [ ! -x $DAEMON ]; then
-  echo "Program not installed or not executable"
-  exit 5
-fi
-
-#
-# init.d / servicectl compatibility (openSUSE)
-#
-if [ -f /etc/rc.status ]; then
-    . /etc/rc.status
-    rc_reset
-fi
-
-#
-# Source function library.
-#
-if [ -f /etc/rc.d/init.d/functions ]; then
-    . /etc/rc.d/init.d/functions
-fi
 
-# overwrite settings from default file
-[ -e /etc/sysconfig/$NAME ] && . /etc/sysconfig/$NAME
+PID_FILE=/var/run/$PROG.pid
+PID_FILE_TIMEOUT=10
 
+DAEMON=/usr/sbin/$PROG
 DAEMON_OPTS="--pidfile=${PID_FILE} --config=${CONF_FILE} cfg:default.paths.data=${DATA_DIR} cfg:default.paths.logs=${LOG_DIR} cfg:default.paths.plugins=${PLUGINS_DIR}"
 
-function isRunning() {
-  status -p $PID_FILE $NAME > /dev/null 2>&1
-}
-
-case "$1" in
-  start)
-    echo -n $"Starting $DESC: .... "
+lockfile=/var/lock/subsys/$PROG
 
-    isRunning
-    if [ $? -eq 0 ]; then
-      echo "Already running."
-      exit 0
-    fi
+# Source the default environment file
```

Comment:
Hi Torkel,

I was modeling the initscript based on EPEL guidelines, which also recommend using lockfiles:
https://fedoraproject.org/wiki/EPEL:SysVInitScript#Initscript_template

I wasn't sure how this should look as part of my proposed change.  The current script is a bit inconsistent:
```
[root@grafana init.d]# service grafana-server start
Starting Grafana Server: .... OK
[root@grafana init.d]# service grafana-server start
Starting Grafana Server: .... Already running.
[root@grafana init.d]# service grafana-server status
grafana-server (pid  10878) is running...
[root@grafana init.d]# service grafana-server stop
OKopping Grafana Server ...                                [  OK  ]
[root@grafana init.d]# service grafana-server stop
Stopping Grafana Server ...(not running)[root@grafana init.d]# 
[root@grafana init.d]# 
[root@grafana init.d]# service grafana-server start
Starting Grafana Server: .... OK
[root@grafana init.d]# service grafana-server restart
OKopping Grafana Server ...                                [  OK  ]
Starting Grafana Server: .... OK
```

My proposed changes look like this:
```
[root@grafana init.d]# service grafana-server start
Starting grafana-server: .                                 [  OK  ]
[root@grafana init.d]# service grafana-server start
[root@grafana init.d]# service grafana-server status
grafana-server (pid  11145) is running...
[root@grafana init.d]# service grafana-server stop
Stopping grafana-server:                                   [  OK  ]
[root@grafana init.d]# service grafana-server stop
[root@grafana init.d]# service grafana-server start
Starting grafana-server: .                                 [  OK  ]
[root@grafana init.d]# service grafana-server restart
Stopping grafana-server:                                   [  OK  ]
Starting grafana-server: .                                 [  OK  ]
```

The question then is how  would you want it to look when starting/stopping after grafana-server has already started/stopped.  My change just exits without putting anything to stdout.

I could leave it blank or change it to any of the following you wish:
```
[root@grafana ~]# service grafana-server start
Starting grafana-server: .... Already running.
Already running.
grafana-server is already running (pid 8277)
```

Similarly, with stopping when grafana-server is not running:
```
Stopping grafana-server: ...(not running)
(not running)
Stopping grafana-server:                                   [FAILED]
```

Let me know which you prefer and I'll make the change.  Thanks!

Giovanni